### ISSUE TITLE
ValueError when running recover_dynamics

### ISSUE THREAD

**Issue by JuliaChristiaanse** [theislab/scvelo#1119]
Hi scvelo people, 

While running scv.tl.recover_dynamics(adata, n_jobs=8) I get the error below:
I used the data & the code from the vignette https://cellrank.readthedocs.io/en/latest/notebooks/tutorials/kernels/200_rna_velocity.html 
but because the error originated from scvelo I decided to try here

my code:
```
adata = sr.datasets.pancreas()
scv.pl.proportions(adata)
scv.pp.filter_and_normalize(
    adata, min_shared_counts=20, n_top_genes=2000, subset_highly_variable=False
)
sc.tl.pca(adata)
sc.pp.neighbors(adata, n_pcs=30, n_neighbors=30, random_state=0)
scv.pp.moments(adata, n_pcs=None, n_neighbors=None)
scv.tl.recover_dynamics(adata, n_jobs=8)
```

My error:
ValueError                                Traceback (most recent call last)
Cell In[5], line 1
----> 1 scv.tl.recover_dynamics(adata, n_jobs=8)

File [c:\Users\julia\.virtualenvs\project-VjJne3mB\lib\site-packages\scvelo\tools\dynamical_model.py:588](file:///C:/Users/julia/.virtualenvs/project-VjJne3mB/lib/site-packages/scvelo/tools/dynamical_model.py:588), in recover_dynamics(data, var_names, n_top_genes, max_iter, assignment_mode, t_max, fit_time, fit_scaling, fit_steady_states, fit_connected_states, fit_basal_transcription, use_raw, load_pars, return_model, plot_results, steady_state_prior, add_key, copy, n_jobs, backend, **kwargs)
    585     adata.varm["loss"] = loss
    587 if t_max is not False:
--> 588     dm = align_dynamics(adata, t_max=t_max, dm=dm, idx=idx)
    590 logg.info("    finished", time=True, end=" " if settings.verbosity > 2 else "\n")
    591 logg.hint(
    592     "added \n"
    593     f"    '{add_key}_pars', "
    594     f"fitted parameters for splicing dynamics (adata.var)"
    595 )

File [c:\Users\julia\.virtualenvs\project-VjJne3mB\lib\site-packages\scvelo\tools\dynamical_model.py:721](file:///C:/Users/julia/.virtualenvs/project-VjJne3mB/lib/site-packages/scvelo/tools/dynamical_model.py:721), in align_dynamics(data, t_max, dm, idx, mode, remove_outliers, copy)
    718 if dm is not None and dm.recoverable:
    719     dm.m = m[idx]
    720     dm.alpha, dm.beta, dm.gamma, dm.pars[:3] = (
--> 721         np.array([dm.alpha, dm.beta, dm.gamma, dm.pars[:3]]) / dm.m[-1]
    722     )
    723     dm.t, dm.tau, dm.t_, dm.pars[4] = (
    724         np.array([dm.t, dm.tau, dm.t_, dm.pars[4]]) * dm.m[-1]
    725     )
    727 return adata if copy else dm

ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (4,) + inhomogeneous part.

my package versions:
scvelo==0.2.5  scanpy==1.9.4  anndata==0.9.2  loompy==3.0.7  numpy==1.24.3  scipy==1.11.2  matplotlib==3.7.1  sklearn==1.3.0  pandas==2.1.0  

I tried to re-install, delete and re-install scvelo in my pipenvironment as per suggestion of cellrank, but this did not solve the issue.
This issue also arises when I use my own dataset.

Any suggestions? 

Thank you in advance :)

**Comment by niehu2018** [comment]
I come to the same problem. How should I work this out?

**Comment by luostrowski** [comment]
I'm having the same problem

**Comment by aditisk** [comment]
Same here, having this issue as well.

**Comment by jwalewski** [comment]
Interesting. I'm having a very similar issue with numpy.array(), although in velocity_graph.py instead. (https://github.com/theislab/scvelo/issues/1121)

**Comment by jwalewski** [comment]
Hey again, I may have found a fix for this issue. For my issue (#1121) I found that by forcing numpy to interpret the array as an object array (by specifying ```numpy.array(data, dtype=object)```) my code proceeded. I am yet to find out if this will work as intended, but just a heads up that there may be a simple fix and I am curious if others facing this similar issue would benefit from modifying the relevant line.

**Comment by JBreunig** [comment]
Same problem here

**Comment by resol341** [comment]
Reverting to pandas==1.5.3 and numpy==1.23.1 seems to have fixed it for me.

**Comment by JuliaChristiaanse** [comment]
Hi guys I solved this bug by directly installing scvelo from the GitHub main repository per recommendation of the CellRank people. Happy coding :)

**Comment by WeilerP** [comment]
Fixed by #1059.
